src/v2/Components/PasswordInput.tsx

```diff
@@ -1,69 +1,52 @@
 import { ClosedEyeIcon, OpenEyeIcon, space } from "@artsy/palette"
-import React from "react"
+import React, { useState } from "react"
 import styled from "styled-components"
-
 import QuickInput, { QuickInputProps } from "./QuickInput"
 
 export interface PasswordInputProps extends QuickInputProps {
   showPasswordMessage?: boolean
 }
 
-export interface PasswordInputState {
-  showPassword: boolean
-}
-
 /**
  * Password input.
  * Renders the label inside of the textbox; shows/hides the password.
  *
  */
-export class PasswordInput extends React.Component<
-  PasswordInputProps,
-  PasswordInputState
-> {
-  state = {
-    showPassword: false,
-  }
-
-  getRightViewForPassword() {
-    const icon = this.state.showPassword ? (
-      <ClosedEyeIcon onClick={this.toggleShowPassword} />
+export const PasswordInput: React.FC<PasswordInputProps> = ({
+  error,
+  showPasswordMessage,
+  ref,
+  ...newProps
+}) => {
+  const [showPassword, toggleShowPassword] = useState(false)
+
+  const eyeIcon = () => {
+    const icon = showPassword ? (
+      <ClosedEyeIcon onClick={() => toggleShowPassword(false)} />
     ) : (
-      <OpenEyeIcon onClick={this.toggleShowPassword} />
+      <OpenEyeIcon onClick={() => toggleShowPassword(true)} />
     )
 
-    return <Eye onClick={this.toggleShowPassword}>{icon}</Eye>
-  }
-
-  toggleShowPassword = () => {
-    this.setState({
-      showPassword: !this.state.showPassword,
-    })
+    return <Eye>{icon}</Eye>
   }
 
-  render() {
-    const { error, showPasswordMessage, ref, ...newProps } = this.props
-
-    const type = this.state.showPassword ? "text" : "password"
-    const note =
-      !error && showPasswordMessage && "Password must be at least 8 characters."
-
-    return (
-      <QuickInput
-        {...newProps}
-        error={error}
-        type={type}
-        rightAddOn={this.getRightViewForPassword()}
-        note={note}
-      />
-    )
-  }
+  const type = showPassword ? "text" : "password"
+  const note =
+    !error && showPasswordMessage && "Password must be at least 8 characters."
```

Comment:
Not especially related to this PR, but this message is a little incomplete. If we're enforcing this client-side, it would be better to match the current [length](https://github.com/artsy/gravity/blob/9a1e70397e8e514ed0b7579fc569c06a08316eee/config/initializers/devise.rb#L121) and [complexity](https://github.com/artsy/gravity/blob/9a1e70397e8e514ed0b7579fc569c06a08316eee/app/models/domain/user.rb#L1096) requirements.

Reply:
We have a ticket for this already, going to save this change -- will need to add a little regex check to get the validation to check for special characters. https://artsyproduct.atlassian.net/browse/PLATFORM-2314